### PR TITLE
chore: testing collator validation missing actual collation

### PR DESCRIPTION
**Pull request by genslein**
We are not actually collating specs from services. Need to debug

image

https://github.com/snyk/vervet/blob/main/vervet-underground/internal/storage/collator.go#L63

We're failing every time

### CODE REVIEW

File: vervet-underground/internal/scraper/scraper_test.go [hunk 1]
```diff
@@ -104,6 +105,28 @@ func TestScraper(t *testing.T) {
...
+		case "2021-10-01":
+			c.Assert(len(spec.Paths), qt.Equals, 3)
+		case "2021-10-16":
+			c.Assert(len(spec.Paths), qt.Equals, 4)
+		case "2021-09-01":
+			c.Assert(len(spec.Paths), qt.Equals, 1)
+		case "2021-09-16":
+			c.Assert(len(spec.Paths), qt.Equals, 2)
+		default:
+			c.FailNow()
```
Comment (cmars) [inline vervet-underground/internal/scraper/scraper_test.go:127]: This would be better as a table test.

File: vervet-underground/internal/scraper/scraper_test.go [hunk 2]
```diff
@@ -211,4 +234,26 @@ func TestScraperCollation(t *testing.T) {
 	collated, err := st.GetCollatedVersionSpecs()
 	c.Assert(err, qt.IsNil)
 	c.Assert(len(collated), qt.Equals, 4)
+
+	c.Assert(len(st.Versions()), qt.Equals, 4)
```
Comment (cmars) [inline vervet-underground/internal/scraper/scraper_test.go:238]: Same here

File: vervet-underground/internal/scraper/scraper.go [hunk 3]
```diff
@@ -148,10 +148,10 @@ func (s *Scraper) scrape(ctx context.Context, scrapeTime time.Time, svc service)
...
 		return errors.WithStack(err)
 	}
-	err = s.storage.NotifyVersions(svc.base, versions, scrapeTime)
-	if err != nil {
-		return errors.WithStack(err)
-	}
+	//err = s.storage.NotifyVersions(svc.base, versions, scrapeTime)
+	//if err != nil {
+	//	return errors.WithStack(err)
+	//}
```
Comment (cmars) [inline vervet-underground/internal/scraper/scraper.go:154]: Why is this removed? I hope it's not because we're trying to debug this in preprod... we'd be better off reproducing in a local test that serves up openapi specs with httptest to be scraped...

File: vervet-underground/internal/scraper/scraper_test.go (hunk 1 above)
Comment (cmars) [reply vervet-underground/internal/scraper/scraper_test.go:127]: Or a map[string]int of "# expected paths" perhaps?

File: vervet-underground/internal/scraper/scraper.go (hunk 3 above)
Comment (genslein) [reply vervet-underground/internal/scraper/scraper.go:154]: Because this actually does nothing and in fact, wipes out local mem when done. We don't have any content since `getNewVersion` hasn't been called so this is effectively not helping at all. Line 166 is where the actual work is being done.

Comment (cmars) [reply vervet-underground/internal/scraper/scraper.go]: Should we just remove it then?

**Review by cmars — APPROVED** [review]

File: vervet-underground/internal/scraper/scraper.go (hunk 3 above)
Comment (cmars) [reply vervet-underground/internal/scraper/scraper.go]: Ok, already was :)

File: vervet-underground/internal/scraper/scraper_test.go (hunk 2 above)
Comment (genslein) [reply vervet-underground/internal/scraper/scraper_test.go:235]: done

File: vervet-underground/internal/scraper/scraper_test.go (hunk 1 above)
Comment (genslein) [reply vervet-underground/internal/scraper/scraper_test.go]: done
